tensorflow_datasets/core/download/download_manager.py

```diff
@@ -336,6 +339,10 @@ def _handle_download_result(
           url=resource.url, url_path=url_path, url_info=url_info)
     elif resource.url not in self._url_infos:
       if self._force_checksums_validation:
+        if tf.io.gfile.exists(self._checksums_path):
```

Comment:
This doesn't seems to be the right place to load the checksums. We should move this when `self._url_infos` is initialised (if `checksums_path ` is set, then load from path, otherwise, load global checksums).

Currently, if the dataset is downloading 1000 images, it will check 1000 times for the path existance.